core-backend/sandbox/src/env.rs

```diff
@@ -171,26 +201,36 @@ impl<E: Ext + IntoExtInfo + 'static> Environment<E> for SandboxEnvironment<E> {
         })
     }
 
-    fn get_wasm_memory_begin_addr(&mut self) -> u64 {
+    fn get_wasm_memory_begin_addr(&self) -> WasmBeginAddress {
         self.runtime.memory.get_wasm_memory_begin_addr()
     }
 
-    fn execute(&mut self, entry_point: &str) -> Result<BackendReport, BackendError> {
+    fn execute<Pre, Post>(
+        mut self,
+        entry_point: &str,
+        _pre_execution_handler: Option<Pre>,
```

Comment:
We can run `Pre` if we've added it I guess.

Reply:
it will be removed. discussed.